manifold/src/constructors.cu

```diff
@@ -268,6 +269,76 @@ Manifold Manifold::Extrude(Polygons crossSection, float height, int nDivisions,
   return extrusion;
 }
 
+Manifold Manifold::Sweep(Polygons crossSection, int steps,
+        std::function<glm::mat4x3(int)> &transform, bool isClosed) {
```

Comment:
So the idea is that transform takes the step int as input and returns a transform matrix? That's not necessarily going to be the easiest function for users to write. Might it be better to just have a 3D polyline as input and deduce the transforms from it? Of course the roll axis gets a little tricky. 

Also it might be possible to simplify this a bit by using extrude/revolve and the warp function - see the Knot example.

Reply:
Yes, my thought is to make this a general implementation and create some helper functions/specialization for the transform function. Generating the transform function from a 3D polyline should be possible?

Reply:
That's fair, but how much different is this version than an `Extrude` followed by a `Warp`? I'd like to see the helper functions and specializations. If this code is more convenient we can always build it inside as an `Impl` method to support them. I'd definitely like to see your 3D polyline version; do you have other ideas as well?

Reply:
The example I give above is a version that is a bit more difficult to create with extrude followed by a warp (perhaps you can specify a twist and calculate the vertices, but this is more complicated)

![image](https://user-images.githubusercontent.com/12198657/166408197-ad2d9671-54a5-4753-9f52-dc71106b592e.png)

For polyline, I am not sure what representation should I use for the polyline, so not much idea for now.

Reply:
Food for thought, this is how I would go about making your spring:
```
const int n = 24;
const int m = 1000;
const float r0 = 10;
const float r1 = 50;
const float h = 300;
const int k = 10;

Polygons circle(1);
float dPhi = 360.0f / n;
for (int i = 0; i < n; ++i) {
  circle[0].push_back(
      {r0 * glm::vec2(cosd(dPhi * i) + 2.0f, sind(dPhi * i)), 0});
}

Manifold spring = Manifold::Extrude(circle, 1, m);

spring.Warp([r1, h, k](glm::vec3& v) {
  const float phi = v.z * k * 2 * glm::pi<float>();
  v.x += v.z * h;
  const float r = v.y + r1;
  v.y = r * glm::cos(phi);
  v.z = r * glm::sin(phi);
});
```

Reply:
A polyline could just be a `std::vector<glm::vec3>`, but the tricky part comes with defining the twist along the axis. Maybe a quaternion at each end?

Reply:
I guess maybe we can have a look at https://github.com/openscad/openscad/pull/2796 ? The way they specify the extrusion seems pretty easy to use.